connector/controllers/main.py

```diff
@@ -60,28 +86,17 @@ def retry_postpone(job, message, seconds=None):
                 return ""
 
         try:
-            # if the job has been manually set to DONE or PENDING,
-            # or if something tries to run a job that is not enqueued
-            # before its execution, stop
-            if job.state != ENQUEUED:
-                _logger.warning('job %s is in state %s '
-                                'instead of enqueued in /runjob',
-                                job_uuid, job.state)
-                return
-
-            with session_hdl.session() as session:
-                # TODO: set_started should be done atomically with
-                #       update queue_job set=state=started
-                #       where state=enqueid and id=
-                job.set_started()
-                self.job_storage_class(session).store(job)
-
-            _logger.debug('%s started', job)
-            with session_hdl.session() as session:
-                job.perform(session)
-                job.set_done()
-                self.job_storage_class(session).store(job)
-            _logger.debug('%s done', job)
+            try:
+                self._try_perform_job(session_hdl, job)
+            except OperationalError as err:
+                # Automatically retry the typical transaction serialization
+                # errors
+                if err.pgcode not in PG_CONCURRENCY_ERRORS_TO_RETRY:
+                    raise
+
+                session.env.clear()
```

Comment:
> Actually, this code might have undesirable side effects.
> 
> Indeed, if we have another env (completely outside the connector) which has fields to recalculate at the same time that our job is failing, then the fields for that env will be cleared too since the todos are shared between envs.

This controller is the entry point of the request. So we should not have other environments than the ones we created here for the handling of the job or for its execution.